robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -166,37 +168,41 @@ def execute(self, userdata=None):
             rospy.loginfo("{} people remaining after None-check".format(len(found_people)))
 
             robot_pose = self._robot.base.get_location()
-            # TODO: Check probable bug here
-            found_people = filter(lambda x: (x.pose.frame.p - robot_pose.frame.p).Norm() < self._look_distance,
-                                  found_people)
+            for z in range(len(list(found_people))):
+                y = list(found_people)[z]
+                rospy.loginfo(" Position of found_people  is {}".format(y.pose))
+                if (y.pose.frame.p - robot_pose.frame.p).Norm() > self._look_distance:
+                    list(found_people).pop(z)
```

Comment:
You created a list from a list and then you pop items. When you create a new list, it does do a deep-copy, empirically determined. So popping here is useless.